CMakeLists.txt

```diff
@@ -38,6 +38,7 @@ option(ICEBERG_BUILD_SHARED "Build shared library" OFF)
 option(ICEBERG_BUILD_TESTS "Build tests" ON)
 option(ICEBERG_ARROW "Build Arrow" ON)
 option(ICEBERG_AVRO "Build Avro" ON)
+option(ICEBERG_IO "Build IO" ON)
```

Comment:
Can we remove this? I think we don't need a separate library for I/O. We can just define `FileIO` and related interfaces in the `libiceberg`. `iceberg-arrow` could possibly implement these APIs by adapting to `arrow::FileSystem`. We can provide an implementation for `local` or `memory` filesystem in the unit test modules.